### PR TITLE
Make the width contain units and add start/end functions

### PR DESCRIPTION
**Pull request by cquiroz**
A Band's width has no units. It should be nanometers

### CODE REVIEW

**Review by rpiaggio — APPROVED** [review]
LGTM

**Review by swalker2m — APPROVED** [review]
Seems good to me.  :+1:  Can't believe we had a loose `Int` in our model like that!

**Comment by cquiroz** [comment]
Yeah. I was a bit surprised too

File: modules/core/shared/src/main/scala/lucuma/core/enum/MagnitudeBand.scala [hunk 1]
```diff
@@ -19,32 +27,40 @@ sealed abstract class MagnitudeBand(
...
   val longName: String,
   val center: Wavelength,
-  val width: Int,
+  val width: Quantity[PosInt, Nanometer],
   val magnitudeSystem: MagnitudeSystem
-) extends Product with Serializable
+) extends Product with Serializable {
+  require(center.toPicometers.value >= width.value / 2)
+
+  def end: Wavelength = Wavelength(center.toPicometers + (width.value.value / 2).withRefinedUnit[Positive, Nanometer])
```
Comment (toddburnside) [inline modules/core/shared/src/main/scala/lucuma/core/enum/MagnitudeBand.scala:35]: Not a big deal here, but I _think_ you can avoid stripping the units off width for the division and then adding them back on by `center.toPicometers + width / 2.withRefinedUnit[Positive, Unitless]`.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/enum/MagnitudeBand.scala:35]: yes you can see
https://github.com/erikerlandson/coulomb/issues/211

**Pull request merged**